src/com/hms/flexyosisoftconnector/OSIsoftServer.java

```diff
@@ -162,31 +176,30 @@ public int setTagWebId(Tag tag) throws JSONException {
       }
       JSONObject requestResponse = new JSONObject(JsonT);
       JSONArray items = new JSONArray();
-      if (requestResponse.has("Items")) {
-        items = requestResponse.getJSONArray("Items");
-      }
+      if (requestResponse.has("Items")) items = requestResponse.getJSONArray("Items");
 
       if (items.length() > 0) {
         // tag exists
-        tag.setWebID(items.getJSONObject(0).getString("WebId"));
+        tagWebID = items.getJSONObject(0).getString("WebId");
       } else {
         // tag does not exist and must be created
+        String payload = PayloadBuilder.buildNewPointBody(tag); // need datapoint? ////

```

Comment:
leftover from debug?
```suggestion
        String payload = PayloadBuilder.buildNewPointBody(tag); 
```